src/classes/RD2_OpportunityService.cls

```diff
@@ -222,53 +240,102 @@ public with sharing class RD2_OpportunityService {
+        if (opp.CloseDate != installment.nextDonationDate) {
+            opp.CloseDate = installment.nextDonationDate;
+            isChanged = true;
+        }
+
+        if (installment.campaignId != null
+            && opp.CampaignId != installment.campaignId
+            && rdSettings.shouldSetInstallmentCampaign
+        ) {
+            opp.CampaignId = installment.campaignId;
+            isChanged = true;
+        }
+
+        return isChanged;
+    }
+
+    /***
+    * @description Updates Opportunity Account and/or Primary Contact if it differs
+    * from the related Recurring Donation. Past open Opportunities are not updated.
+    * @param opp Opportunity
     * @param rd Recurring Donation
-    * @return Boolean
+    * @return Boolean Indicates if the donor info on the Opportunity has been changed
     */
-    private Boolean syncOpportunity(Opportunity opp, RD2_ScheduleService.Installment installment, npe03__Recurring_Donation__c rd) {  
+    private Boolean syncOppDonorInfo(Opportunity opp, npe03__Recurring_Donation__c rd) {
+        if (opp.CloseDate < currentDate) {
+            return false;
+        }
+
         Boolean isChanged = false;
 
         if (opp.AccountId != rd.npe03__Organization__c) {
             opp.AccountId = rd.npe03__Organization__c;
             isChanged = true;
         }
-        
+
         if (opp.Primary_Contact__c != rd.npe03__Contact__c) {
             opp.Primary_Contact__c = rd.npe03__Contact__c;
             isChanged = true;
         }
+        
+        return isChanged;
+    }
 
-        if (installment != null) { 
-            if (opp.Amount != installment.installmentAmount) {
-                opp.Amount = installment.installmentAmount;
-                isChanged = true;
-            }
-
-            if (opp.CloseDate != installment.nextDonationDate) {
-                opp.CloseDate = installment.nextDonationDate;
-                isChanged = true;
-            }
-
-            if (installment.campaignId != null 
-                && opp.CampaignId != installment.campaignId 
-                && rdSettings.shouldSetInstallmentCampaign
-            ) {
-                opp.CampaignId = installment.campaignId;
-                isChanged = true;
-            }
+    /***
+    * @description Updates Opportunity CurrencyIsoCode if it differs from the related Recurring Donation 
+    * @param opp Opportunity
+    * @param rd Recurring Donation
+    * @return Boolean Indicates if the currency on the Opportunity has been changed
+    */
+    private Boolean syncOppCurrency(Opportunity opp, npe03__Recurring_Donation__c rd) {
+        if (!isMultiCurrencyEnabled) {
+            return false;
         }
 
-        if (isChanged) {
-            opp.name = buildOpportunityName(rd.npe03__Donor_Name__c, opp);
+        String rdCurrency = (String) rd.get(CURRENCY_ISO_CODE_FIELD);
+        if ((String) opp.get(CURRENCY_ISO_CODE_FIELD) != rdCurrency) {
+            opp.put(CURRENCY_ISO_CODE_FIELD, rdCurrency);
```

Comment:
If I'm not mistaken, there's an issue with how Salesforce handles this kind of update. If the CurrencyCode is changed it will convert the Amount (i.e. change it). The way we handled this on the Allocation and Payment objects was to always update the CurrencyCode and the Amount together. That doesn't have to happen in this method, but it might be necessary to ensure that the Amount is updated on the Opp even if technically the Amount value wasn't changed on the RD. Not sure I'll have time to test this in the UI today to see if happens here though.

Reply:
I cannot reproduce the issue via unit tests or UI. Can you please let me know details.

Reply:
@Julian88Tex , @ravensisco  ^